torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -264,6 +265,29 @@ def __init__(
         # Turn on if compression / decompression computation is a bottleneck.
         self.batch_tensors_with_same_shape = batch_tensors_with_same_shape
 
+    def __getstate__(self):
+        r"""
+        Returns a ``Dict[str, Any]`` which will be pickled and saved.
+        ``process_group`` is not serializable and excluded from
+        a returned state.
+        """
+        return {
+            slot: getattr(self, slot)
+            for slot in self.__slots__ if slot != "process_group"
+        }
+
+    def __setstate__(self, state):
+        r"""
+        Takes a provided ``state`` and retrieves ``PowerSGDState``.
+        ``process_group`` is set to default.
+        """
+        self.process_group = distributed_c10d._get_default_group()
+        logger.warning(
+            "NOTE: Current process group is set to default."
```

Comment:
nit: Change the warning to something like

`Process group will be set to default group (whole world) when loading PowerSGD state`. If a different group is desired, please set `self.process_group` after PowerSGD state is loaded`.

Reply:
Warning message is updated